src/ql/com/tests/topology_test.cc

```diff
@@ -0,0 +1,156 @@
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
+#include "doctest/doctest.h"
+
+#include "ql/com/topology.h"
+
+namespace ql {
+namespace com {
+
+class TopologyTest {};
+
+TEST_CASE_FIXTURE(TopologyTest, "Single core grid 2x3") {
```

Comment:
Since this class does a lot of things, including user input validation, there are still a lot of unit tests missing for e.g. checking that exceptions are thrown properly. Not in this change though.